language/move-lang/tests/functional/shared_ed25519_public_key/shared_key.move

```diff
@@ -0,0 +1,31 @@
+// Test that publishing and rotating the shared key works as expected.
+// It is tricky to test that a transaction can still be sent after key rotation in a functional
+// test, so that case is covered in an e2e test.
+
+use 0x0::LibraAccount;
+use 0x0::SharedEd25519PublicKey;
+use 0x0::Transaction;
+fun main() {
+    let old_auth_key = LibraAccount::authentication_key({{default}});
+    let pubkey1 = x"0000000000000000000000000000000000000000000000000000000000000000";
+    SharedEd25519PublicKey::publish(copy pubkey1);
+    let new_auth_key = LibraAccount::authentication_key({{default}});
+
+    // check that publishing worked
+    Transaction::assert(SharedEd25519PublicKey::exists({{default}}), 3000);
+    Transaction::assert(SharedEd25519PublicKey::key({{default}}) == pubkey1, 3001);
```

Comment:
Is it always true that `SharedEd25519PublicKey::key({{default}}) == LibraAccount::authentication_key({{default}})`?

Reply:
Not quite--it is always true that `Authenticator::ed25519_authentication_key(SharedEd25519PublicKey::key({{default}})) == LibraAccount::authentication_key({{default}})` (that is, the shared key is always the preimage of the auth key). Would definitely be happy to use a name other than "shared key" if there is something  that conveys this better!

Reply:
I see, makes sense now, thank you!